lib/plsql/procedure_call.rb

```diff
@@ -236,6 +236,14 @@ def add_argument(argument, value, argument_metadata=nil)
         @bind_values[argument] = value.nil? ? nil : (value ? 1 : 0)
         @bind_metadata[argument] = argument_metadata.merge(:data_type => "NUMBER", :data_precision => 1)
         "l_#{argument}"
+      when 'UNDEFINED'
+        if argument_metadata[:type_name] == 'XMLTYPE'
+          @declare_sql << "l_#{argument} XMLTYPE;\n"
+          @assignment_sql << "l_#{argument} := XMLTYPE(:#{argument});\n" if not value.nil?
+          @bind_values[argument] = value if not value.nil?
+          @bind_metadata[argument] = argument_metadata.merge(:data_type => "CLOB")
+          "l_#{argument}"
+        end
```

Comment:
Regardless of the change introduced the unsupported datatypes (like ANYDATA) will not work.
But you're right that there is a change in the evaluation for UNDEFINED datatypes.
This can be changed so the rest of UNDEFINED fall into the behaviour prior to the change.
I'm not sure what would be the difference since one way or the other the types like ANYDATA are not supported.